src/OrchardCore/OrchardCore.Taxonomies.Core/DefaultContentsTaxonomyListQueryService.cs

```diff
@@ -32,10 +32,18 @@ public async Task<IQuery<ContentItem>> QueryAsync(TermPart termPart, Pager pager
 
         await _contentTaxonomyListFilters.InvokeAsync((filter, q, part) => filter.FilterAsync(q, part), query, termPart, _logger);
 
+        var currentPage = 1;
+
+        if (pager?.Page > 0)
```

Comment:
If `pager` is `null` here, then there will be an NRE on line 46 anyway. Better add an `ArgumentNullException.ThrowIfNull(pager);` to the top.